src/test/java/pages/BasePage.java

```diff
@@ -0,0 +1,55 @@
+package pages;
+import io.github.bonigarcia.wdm.WebDriverManager;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.chrome.ChromeOptions;
+import org.openqa.selenium.interactions.Actions;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.openqa.selenium.support.PageFactory;
+import org.openqa.selenium.By;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.BeforeSuite;
+import org.testng.annotations.Parameters;
+import java.time.Duration;
+public class BasePage {
+    protected WebDriver driver;
+    protected WebDriverWait wait;
+    protected Actions actions;
+    String url;
+    public BasePage(WebDriver givenDriver) {
+        driver = givenDriver;
+        wait = new WebDriverWait(driver, Duration.ofSeconds(5));
+        actions = new Actions(driver);
+        PageFactory.initElements(driver,this);
+    }
+    public void click(By locator) {
+        wait.until(ExpectedConditions.elementToBeClickable(locator));
+    }
+    public WebElement findElement (By locator) {
+        return wait.until(ExpectedConditions.visibilityOfElementLocated(locator));
+    }
+    public void contextClick(By locator) {
+        actions.contextClick(findElement(locator)).perform();
+    }
+    public void doubleClick (By locator) {
+        actions.doubleClick(findElement(locator)).perform();
+    }
+
+    @BeforeSuite
+    static void setupClass() {
+        WebDriverManager.chromedriver().setup();
+    }
+    @BeforeMethod
+    @Parameters({"BaseUrl"})
+    public void launchBrowser(String BaseUrl) {
+        ChromeOptions options = new ChromeOptions();
+        options.addArguments("--disable-notifications", "--remote-allow-origins=*", "--incognito", "--start-maximized");
+        driver = new ChromeDriver(options);
+        driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(10));
+        url = BaseUrl;
+        driver.get(BaseUrl);
+        wait = new WebDriverWait(driver, Duration.ofSeconds(4));
```

Comment:
Base Page \ Base Class — Implements the necessary methods for work with webdriver.

Please move this code in class BaseTest

![Screenshot_1](https://user-images.githubusercontent.com/104761482/232979221-9c4a7200-d1ca-4bdd-941b-3008949f8cb3.png)